### PR TITLE
Update hammer-cli options for 6.11 

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
**Description:**
1. Update `hammer-cli` options for 6.11 in hammer_commands.json, generated from running `scripts/hammer_command_tree.py`
2. Fixing `scripts/hammer_command_tree.py` by adding a hostname arg to ssh.command for runnning locally.
3. Removing `.splitlines()` from parse_help helper, which was added after moving from paramiko to ssh2-python.
4. Fixing `cli/test_hammer.py` which wasn't returning all available differences earlier, here's all available differences,

```
>           pytest.fail(format_commands_diff(differences))
E           Failed: hammer ansible roles
E             Added subcommands:
E               * sync
E
E           hammer ansible roles sync (new command)
E             Added options:
E               * proxy-id
E               * organization-id
E               * organization
E               * location-id
E               * location
E               * organization-title
E               * help
E               * role-names
E               * location-title
E
E           hammer architecture create
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer architecture delete
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer architecture info
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer architecture list
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer architecture update
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer bookmark create
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer bookmark delete
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer bookmark info
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer bookmark list
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer bookmark update
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer capsule
E             Removed subcommands:    * import-classes
E           hammer capsule content synchronize
E             Added options:
E               * content-view-id
E               * repository
E               * content-view
E               * repository-id
E
E           hammer capsule info
E             Added options:
E               * include-status
E               * include-version
E
E           hammer capsule list
E             Added options:
E               * include-status
E
E           hammer compute-profile create
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer compute-profile delete
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer compute-profile info
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer compute-profile list
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer compute-profile update
E             Added options:
E               * organization-id
E               * organization
E               * location-id
E               * organization-title
E               * location
E               * location-title
E
E           hammer compute-resource associate-vms
E             Added options:
E               * vm-id
E
E           hammer content-units (new command)
E             Added subcommands:
E               * list
E               * info
E             Added options:
E               * help
E
E           hammer content-units info (new command)
E             Added options:
E               * content-view-id
E               * repository
E               * organization-id
E               * organization
E               * content-view-version-id
E               * organization-title
E               * product-id
E               * name
E               * organization-label
E               * product
E               * fields
E               * content-type
E               * id
E               * repository-id
E               * content-view
E               * help
E               * content-view-version
E
E           hammer content-units list (new command)
E             Added options:
E               * content-view-id
E               * lifecycle-environment-id
E               * repository
E               * full-result
E               * organization-id
E               * organization
E               * include-filter-ids
E               * content-view-filter-id
E               * content-view-version-id
E               * organization-title
E               * product-id
E               * organization-label
E               * per-page
E               * order
E               * ids
E               * page
E               * search
E               * product
E               * content-view-filter-rule-id
E               * content-view-filter-rule
E               * fields
E               * content-type
E               * help
E               * repository-id
E               * content-view
E               * content-view-filter
E               * content-view-version
E
E           hammer content-view publish
E             Added options:
E               * is-force-promote
E               * lifecycle-environment-ids
E               * lifecycle-environments
E
E           hammer content-view remove
E             Added options:
E               * destroy-content-view
E             Removed options:
E               * lifecycle-environments
E
E           hammer content-view version republish-repositories
E             Added options:
E               * force
E
E           hammer deb-package list
E             Added options:
E               * host
E               * packages-restrict-upgradable
E               * packages-restrict-applicable
E               * host-id
E               * available-for
E             Removed options:
E               * include-filter-ids
E               * content-view-filter-rule
E               * content-view-filter-rule-id
E               * lifecycle-environment
```
**Confirmed these commands/options with the following upstream PR's included in the latest snap of 6.11**
Return org and loc options to all hammer commands https://github.com/theforeman/hammer-cli-foreman/pull/575
Associate a VM via hammer to a compute resource  https://github.com/theforeman/hammer-cli-foreman/pull/577
Add Generic Content Unit Support https://github.com/Katello/hammer-cli-katello/pull/826
Make katello commands work without Puppet https://github.com/Katello/hammer-cli-katello/pull/818
Ansible sync roles & variables https://github.com/theforeman/hammer-cli-foreman-ansible/pull/6/
Add proxy status and version to info command https://github.com/theforeman/hammer-cli-foreman/pull/572
Add Deb Package applicability  https://github.com/Katello/katello/pull/8277
Delete CV https://github.com/Katello/katello/pull/9660
Add force option to metadata regeneration(republish-repositories) https://github.com/Katello/katello/pull/9654
Expose ability to sync individual content view and repository to smart proxy https://github.com/Katello/katello/pull/9490
Add promotion array to publish API endpoint https://github.com/Katello/katello/pull/9352

Signed-off-by: Gaurav Talreja <gtalreja@redhat.com>

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
**Test Results:**
```
(.robottelo) ➜  robottelo git:(fix-hammer-options70) ✗ pytest -q --disable-pytest-warnings tests/foreman/cli/test_hammer.py -k test_positive_all_options
.                                                                                                                                 [100%]
1 passed, 2 deselected in 24.06s
```

File: scripts/hammer_command_tree.py [hunk 1]
```diff
@@ -6,14 +6,14 @@
...
+from robottelo.config import settings
 
 
 def generate_command_tree(command):
     """Recursively walk trhough the hammer commands and subcommands and fetch
     their help. Return a dictionary with the contents.
-
     """
-    output = ssh.command(f'{command} --help').stdout
+    output = ssh.command(f'{command} --help', hostname=settings.server.hostnames[0]).stdout
```
Comment (jyejare) [inline scripts/hammer_command_tree.py:16]: This should be `settings.server.hostname` but you need to explicitly pass this, the `ssh.command` is smart enough to fetch it if you dont provide the hostname.

**Review by jyejare — COMMENTED** [review]
ACK pending hostname param to `ssh.command` not required comment.

Also, the json file is too big that its not visible, could you put that somewhere in private bin and share on a group to view ?

File: scripts/hammer_command_tree.py (hunk 1 above)
Comment (swadeley) [reply scripts/hammer_command_tree.py:16]: Hi @jyejare , reading " but you **need** to explicitly pass this" and then " smart enough to fetch it if you dont provide" makes me wonder if you meant " but you **don't need** to explicitly pass this"?

Comment (Gauravtalreja1) [reply scripts/hammer_command_tree.py:16]: @jyejare we set `settings.server.hostname` in [pytest_fixtures/xdist.py](https://github.com/SatelliteQE/robottelo/blob/master/pytest_fixtures/core/xdist.py#L32), which is in the scope of pytest, and this is just a Python script which is to run locally.
So for `ssh.command` here we need `hostname` param to point to the satellite to check available commands.

and yes, git does not render large diffs by default, but you could check raw file to see that json file https://raw.githubusercontent.com/SatelliteQE/robottelo/a50a99398e81e1cd6d3e421ee700710ba74f879f/tests/foreman/data/hammer_commands.json

Comment (jyejare) [reply scripts/hammer_command_tree.py:16]: @swadeley I mean to say he need *not* to pass hostname explicitely !

Comment (jyejare) [reply scripts/hammer_command_tree.py:16]: @Gauravtalreja1 Ohh, if this script doesn't run under the pytest session scope, are you passing the satellite server hostname in `conf/server.yaml` manually ?

Comment (Gauravtalreja1) [reply scripts/hammer_command_tree.py:16]: @jyejare Yes, you're correct, we run this script once per y-stream to generate a new hammer_commands.json, which will have updated commands/options in it.

**Review by jyejare — APPROVED** [review]

**Review by omkarkhatavkar — APPROVED** [review]

**Pull request merged**